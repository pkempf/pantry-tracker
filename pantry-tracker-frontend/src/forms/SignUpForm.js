import React, { useContext } from "react";
import { Redirect } from "react-router-dom";
import Container from "react-bootstrap/Container";
import PantryForm from "./PantryForm";
import UserContext from "../UserContext";

const SignUpForm = ({ signUpFunction }) => {
  const user = useContext(UserContext);

  if (user.username) {
    return <Redirect to="/" />;
  }

  const formFields = [
    {
      name: "username",
      label: "Username",
      inputType: "text",
      placeholder: "Username",
      initialValue: "",
      required: true,
    },
    {
      name: "password",
      label: "Password",
      inputType: "password",
      placeholder: "Password",
      initialValue: "",
      required: true,
    },
    {
      name: "firstName",
      label: "First Name",
      inputType: "text",
      placeholder: "First",
      initialValue: "",
      required: true,
    },
    {
      name: "lastName",
      label: "Last Name",
      inputType: "text",
      placeholder: "Last",
      initialValue: "",
      required: true,
    },
    {
      name: "email",
      label: "Email",
      inputType: "email",
      placeholder: "(e.g. handle@website.com)",
      initialValue: "",
      required: true,
    },
  ];

  return (
    <Container className="SignUpForm justify-content-center">
      <PantryForm
        formTitle="Register for Pantry Tracker"
        fields={formFields}
        submitButtonText="Sign up"
        processData={signUpFunction}
      />
    </Container>
  );
};

export default SignUpForm;
