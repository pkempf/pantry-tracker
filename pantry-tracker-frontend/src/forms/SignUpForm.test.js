import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SignUpForm from "./SignUpForm";
import UserContext from "../UserContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{}}>
        <SignUpForm />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{}}>
        <SignUpForm />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
