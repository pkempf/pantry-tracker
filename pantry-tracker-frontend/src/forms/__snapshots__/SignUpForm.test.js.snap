// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignUpForm justify-content-center container"
  >
    <div
      class="PantryForm justify-content-center row"
    >
      <div
        class="col-md-8 col-sm-10"
      >
        <div
          class="card"
        >
          <h4
            class="card-header"
          >
            Register for Pantry Tracker
          </h4>
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="username"
                >
                  Username
                </label>
                <input
                  class="form-control"
                  id="username"
                  name="username"
                  placeholder="Username"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="password"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="firstName"
                >
                  First Name
                </label>
                <input
                  class="form-control"
                  id="firstName"
                  name="firstName"
                  placeholder="First"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="lastName"
                >
                  Last Name
                </label>
                <input
                  class="form-control"
                  id="lastName"
                  name="lastName"
                  placeholder="Last"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="email"
                >
                  Email
                </label>
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="(e.g. handle@website.com)"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <button
                class="mr-2 btn btn-primary"
                type="submit"
              >
                Sign up
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
